app/assets/javascripts/actions/story.js

```diff
@@ -15,21 +15,36 @@ export const updateStorySuccess = (story) => ({
   story
 });
 
-export const updateStory = (story, projectId) => {
-  return (dispatch, getState, { Story }) => {
+export const deleteStorySuccess = (id) => ({
+  type: actionTypes.DELETE_STORY_SUCCESS,
+  id
+});
+
+export const editStory = (id, newAttributes) => ({
+  type: actionTypes.EDIT_STORY,
+  id,
+  newAttributes
+});
+
+export const updateStory = (story, projectId) =>
+  (dispatch, getState, { Story }) => {
     if (story._editing._isDirty) {
       return Story.update(story._editing, projectId)
-        .then(( story ) => {
+        .then((story) => {
           dispatch(updateStorySuccess(story));
           dispatch(toggleStory(story.id));
         });
     }
     return dispatch(toggleStory(story.id));
   };
-};
 
-export const editStory = (id, newAttributes) => ({
-  type: actionTypes.EDIT_STORY,
-  id,
-  newAttributes
-});
+export const deleteStory = (storyId, projectId) =>
+  (dispatch, getState, { Story }) =>
+    Story.deleteStory(storyId, projectId)
+      .then(dispatch(deleteStorySuccess(storyId)))
```

Comment:
Since this needs to be dispatched just _after_ the request, you gotta wrap it with a lambda, otherwise it will just be called right away:

```suggestion
      .then(() => dispatch(deleteStorySuccess(storyId)))
```